storage/store.go

```diff
@@ -2703,11 +2703,21 @@ func (s *Store) canApplySnapshotLocked(rangeID roachpb.RangeID, snap raftpb.Snap
 		// When such a conflict exists, it will be resolved by one range
 		// either being split or garbage collected.
 		exReplica, err := s.getReplicaLocked(exRange.Desc().RangeID)
+		msg := "snapshot intersects existing range"
 		if err != nil {
 			log.Warning(context.TODO(), errors.Wrapf(
 				err, "unable to look up overlapping replica on %s", exReplica))
+		} else {
+			exLease, exPendingLease := exReplica.getLease()
+			now := s.Clock().Now()
+			// If the RHS shows no signs of recent activity, give it a GC run.
```

Comment:
Why bother checking the lease? This kind of situation seems rare enough it would just be less complication and effectively the same thing to add the replica to the GC queue and be done with it.